src/components/pages/devices/flyouts/deviceJobs/deviceJobProperties.js

```diff
@@ -1,20 +1,298 @@
+
+    // Linked components
+    this.jobNameLink = this.linkTo('jobName')
+      .reject(nonAlphaNumeric)
+      .check(Validator.notEmpty, () => this.props.t('devices.flyouts.jobs.validation.required'));
+
+    this.propertiesLink = this.linkTo('commonProperties');
+  }
+
+  componentDidMount() {
+    if (this.props.devices) {
+      this.populateState(this.props.devices);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.devices && (this.props.devices || []).length !== nextProps.devices.length) {
+      this.populateState(nextProps.devices);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.populateStateSubscription) this.populateStateSubscription.unsubscribe();
+    if (this.submitJobSubscription) this.submitJobSubscription.unsubscribe();
+  }
+
+  populateState(devices) {
+    if (this.populateStateSubscription) this.populateStateSubscription.unsubscribe();
+
+    // Rework device data so the reported and desired values are grouped together under the property name.
+    const devicesWithProps = [];
+    devices.forEach(device => {
+      const properties = {};
+      Object.keys(device.properties).forEach((propertyName) => {
+        const reported = device.properties[propertyName];
+        const desired = device.desiredProperties[propertyName];
+        const inSync = !desired || reported === desired;
+        const display = inSync ? reported : `${reported} (${this.props.t('devices.flyouts.jobs.properties.syncing')} ${desired})`
+        properties[propertyName] = { reported, desired, display, inSync };
+      });
+      devicesWithProps.push({ id: device.id, properties });
+    });
+
+    this.populateStateSubscription = Observable.from(devicesWithProps)
+      .map(({ properties }) => new Set(Object.keys(properties)))
+      .reduce((commonProperties, deviceProperties) =>
+        commonProperties
+          ? new Set([...commonProperties].filter(property => deviceProperties.has(property)))
+          : deviceProperties
+      ) // At this point, a stream of a single event. A common set of properties.
+      .flatMap(commonPropertiesSet =>
+        Observable.from(devicesWithProps)
+          .flatMap(({ properties }) => Object.entries(properties))
+          .filter(([property]) => commonPropertiesSet.has(property))
+      )
+      .distinct(([propertyName, propertyVal]) => `${propertyName} ${propertyVal.display}`)
+      .reduce((acc, [propertyName, propertyVal]) => update(acc, {
+        [propertyName]: {
+          $autoArray: {
+            $push: [propertyVal]
+          }
+        }
+      }), {})
+      .flatMap(propertyToValMap => Object.entries(propertyToValMap))
+      .reduce(
+        (newState, [name, values]) => {
+          const valueTemp = values.length === 1 ? values[0].display : propertyJobConstants.multipleValues;
+          const type = values.every(({ reported }) => isNumeric(reported)) ? propertyJobConstants.numberType : propertyJobConstants.stringType;
```

Comment:
I won't block this PR but make a TODO to unify these `every` methods into a single reduce to avoid looping multiple times.